### PR TITLE
[DD-273] Scroll Depth Progress Bar

### PR DESCRIPTION
**Pull request by nramabad**
[Preview](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/docs-disco/scroll-depth-progress-bar-preview/?force-all)

[DD-273](https://circleci.atlassian.net/browse/DD-273)

**What and Why?**

Show the progress bar indicating how far the reader has finished reading so the user feels encouraged to finish reading the article

Impact: Encourage the readers to finish reading the article by gamification

**Definition of Done/Acceptance Criteria:** 

Refer to [design](https://www.figma.com/file/HrgpUU2M3ED7OT0p0f99bH/Docs-Disco-Experiments?node-id=4156%3A8738)

[Live Example](https://learn.hashicorp.com/tutorials/terraform/state-cli)

Calculate progress based on article progress (scrolling, similar to scroll highlighting change)

Design review with  @danbeere

### CODE REVIEW

**Comment by tshen815** [comment]
Do we want the progress bar to show up on the landing page as well for docs or only for the articles?

**Comment by nramabad** [comment]
> Do we want the progress bar to show up on the landing page as well or only for the articles?

sounds like a question for @emmasharma-circleci @danbeere ?

**Review by tshen815 — APPROVED** [review]
Looks good on my end, great work!

File: src/js/services/progressbar.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { init } from './progressbar';
+
+describe('Progress Bar', () => {
```
Comment (ccgeller) [inline src/js/services/progressbar.test.js:3]: wooo tests love it!

File: src/styles/_progressbar.scss [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    background: linear-gradient(-155deg, #fff 50%, #008647 50%);
+    background-size: 100% calc(100% - 100vh + 74px);
+    background-repeat: no-repeat;
+
+    @media (max-width: $screen-md) {
+        background: linear-gradient(-140deg, #fff 50%, #008647 50%);
+    }
+}
+
+body::before {
```
Comment (ccgeller) [inline src/styles/_progressbar.scss:34]: is there a way to use 'global-nav--component' instead of boxy?

File: src/styles/main.scss [hunk 3]
```diff
@@ -66,7 +66,7 @@ div.main-body{
 
 div.main-body {
 	display: flex;
-	background: #fff;
+	// background: #fff;
```
Comment (ccgeller) [inline src/styles/main.scss:69]:
```suggestion
```

File: src/styles/main.scss [hunk 4]
```diff
@@ -205,7 +205,7 @@ div.article-container{
     @media (max-width: $screen-sm) {
       width: 100%;
     }
-    min-height: calc(100vh - 68px); // make sure content always fills available space (68px is header size)
+    min-height: calc(100vh - 70px); // make sure content always fills available space (70px is header size)
```
Comment (ccgeller) [inline src/styles/main.scss:208]: Can we make the header size a variable? Then don't have to manually updated in each place

**Review by ccgeller — APPROVED** [review]
Works for me! Great job! Woo Love the tests! (Just minor comments nothing blocking) I think that updating the index files as in this [PR](https://github.com/circleci/circleci-docs/pull/6061/files ): will fix preview issue
Screen Shot 2021-12-21 at 4 06 25 PM

File: src/styles/_progressbar.scss (hunk 2 above)
Comment (nramabad) [reply src/styles/_progressbar.scss:34]: they perform to different functions.

**Pull request merged**